source/slang/slang-check-conversion.cpp

```diff
@@ -547,8 +673,27 @@ namespace Slang
            !canCoerce(toType, fromInitializerListExpr->type, nullptr))
             return _failedCoercion(toType, outToExpr, fromInitializerListExpr);
 
+        // try to invoke the user-defined constructor if it exists
+        if(_invokeExprForExplicitCtor(toType, fromInitializerListExpr, outToExpr))
+        {
+            return true;
+        }
+
+        // try to invoke the synthesized constructor if it exists
+        if (!_invokeExprForSynthesizedCtor(toType, fromInitializerListExpr, outToExpr))
+        {
+            // TODO: check if it's C-style initializer list
+            // if it's not, return error.
+            // if it is, return, fall-back to legacy logic
+            getSink()->diagnoseRaw(Severity::Error, "Synthesized Constructor is not found\n");
+            return false;
+        }
+
+        // We will fall back to the legacy logic of initialize list.
         if(!_readAggregateValueFromInitializerList(toType, outToExpr, fromInitializerListExpr, argIndex))
+        {
```

Comment:
`_readAggregateValueFromInitializerList` needs to be rewritten to turn into something that forms the argument list for the synthesized ctor call.

`_coerceInitializerList` should always set `outToExpr` to an invokeExpr of some ctor.

Reply:
But this can be cleanly separated out into its own PR.

Reply:
Yea. 
You may notice in my later commits of this PR, I don't change `_readAggregateValueFromInitializerList` this function at all, as I want to keep this function as it for the legacy logic only.